app/Http/Controllers/Admin/AdminReportController.php

```diff
@@ -1327,4 +1349,173 @@ public function reportsApiSpamGet(Request $request, $id)
 
         return new AdminSpamReport($report);
     }
+
+    public function reportsApiRemoteHandle(Request $request)
+    {
+        $this->validate($request, [
+            'id' => 'required|exists:remote_reports,id',
+            'action' => 'required|in:mark-read,cw-posts,unlist-posts,delete-posts,private-posts,mark-all-read-by-domain,mark-all-read-by-username,cw-all-posts,private-all-posts,unlist-all-posts'
+        ]);
+
+        $report = RemoteReport::findOrFail($request->input('id'));
+        $user = User::whereProfileId($report->account_id)->first();
+        $ogPublicStatuses = [];
+        $ogUnlistedStatuses = [];
+        $ogNonCwStatuses = [];
+
+        switch ($request->input('action')) {
+            case 'mark-read':
+                $report->action_taken_at = now();
+                $report->save();
+                break;
+            case 'mark-all-read-by-domain':
+                RemoteReport::whereInstanceId($report->instance_id)->update(['action_taken_at' => now()]);
```

Comment:
If you really want to ignore all reports from a domain, which should really be a last-resort action, then you'd want to handle this via Federation Management, whereby you reject all incoming reports from that instance.

it may make sense to be able to filter reports by instance, such that you can have a queue to easily process. The side effect of having a bulk action like this is that you might throw out a good report with all the bad reports